wallet-new/src/Cardano/Wallet/API/V1/LegacyHandlers/Wallets.hs

```diff
@@ -186,3 +198,95 @@ updateWallet wid WalletUpdate{..} = do
         -- reacquire the snapshot because we did an update
         ws' <- V0.askWalletSnapshot
         addWalletInfo ws' updated
+
+-- | Creates a new or restores an existing external @wallet@ given a 'NewExternalWallet' payload.
+-- Returns to the client the representation of the created or restored wallet in the 'Wallet' type.
+newExternalWallet
+    :: ( MonadThrow m
+       , V0.MonadBlockchainInfo m
+       , V0.MonadWalletLogic ctx m
+       )
+    => NewExternalWallet
+    -> m (WalletResponse Wallet)
+newExternalWallet NewExternalWallet{..} = do
+    let newWalletHandler CreateWallet  = createNewExternalWallet
+        newWalletHandler RestoreWallet = error "Restore external wallet, TODO"
```

Comment:
Replace with `undefined` and a comment so that it won't get accidentally left in as an intentional `error` :)